0007-onion-routing-over-libp2p.adoc

```diff
@@ -0,0 +1,95 @@
+= Improving privacy through onion routing over libp2p
+Daniel Karzel <daniel.karzel@coblox.tech>;
+:toc:
+:revdate: 2019-06-17
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/coblox/spikes/issues/8[Tor Support for COMIT]
+
+== Context
+
+Onion routing as implemented by the https://en.wikipedia.org/wiki/Tor_(anonymity_network)[Tor network] enables anonymous usage of the internet through end2end multi-time-encryption over multiple hops provided through relay nodes.
+Anybody in the network can set up a relay node.
+
+The goal of an anonymous COMIT is to be able to run the comit-node in a way, so that the person running the comit-node cannot be discovered and it cannot be discovered with whom that person exchanges information over COMIT.
+Note that a person might not be concerned about the blockchain nodes, but only about COMIT.
+
+Since anonymity networks like Tor build on the fact that there are many people using it, it would not make much sense to just make the COMIT network anonymous by itself but rather channel the transport over an existing anonymity network like e.g. Tor.
```

Comment:
```suggestion
Since anonymity networks like Tor build on the fact that there are many people using it, it would not make much sense to just make the COMIT network anonymous by itself but rather channel the transport over an existing anonymity network e.g. Tor.
```

Reply:
fun fact: I just edited your comment and then committed it :D